### PR TITLE
Added ui config, compute, storage and IAM apis

### PR DESCRIPTION
**Pull request by risha-vijayvargiya-22**
The PR contains:

1. UI config API
2. Region API
3. Network API
4. Sub Network API
5. Shared Network API
6. Cloud Storage Bucket API
7. Service Account API
8. GetXPNHost API

### CODE REVIEW

File: dataproc_jupyter_plugin/controllers/vertex/cloudStorageBucket.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import aiohttp
+import tornado
+from jupyter_server.base.handlers import APIHandler
+
+from dataproc_jupyter_plugin import credentials
+from dataproc_jupyter_plugin.services.vertex import vertex
+
+
+class CloudStorageBucketController(APIHandler):
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/controllers/vertex/cloudStorageBucket.py]: Change the name to CloudStorageController and move the class to the storage.py file

File: dataproc_jupyter_plugin/controllers/vertex/network.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import aiohttp
+import tornado
+from jupyter_server.base.handlers import APIHandler
+
+from dataproc_jupyter_plugin import credentials
+from dataproc_jupyter_plugin.services.vertex import vertex
+
+
+class NetworkController(APIHandler):
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/controllers/vertex/network.py]: Move this method to a file called compute.py

File: dataproc_jupyter_plugin/controllers/vertex/region.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import aiohttp
+import tornado
+from jupyter_server.base.handlers import APIHandler
+
+from dataproc_jupyter_plugin import credentials
+from dataproc_jupyter_plugin.services.vertex import vertex
+
+
+class RegionController(APIHandler):
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/controllers/vertex/region.py]: Move to compute.py

File: dataproc_jupyter_plugin/controllers/vertex/sharedNetwork.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+import aiohttp
+import tornado
+from jupyter_server.base.handlers import APIHandler
+
+from dataproc_jupyter_plugin import credentials
+from dataproc_jupyter_plugin.services.vertex import vertex
+
+
+class SharedNetworkController(APIHandler):
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/controllers/vertex/sharedNetwork.py]: Move to compute.py

File: dataproc_jupyter_plugin/controllers/vertex/subNetwork.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import aiohttp
+import tornado
+from jupyter_server.base.handlers import APIHandler
+
+from dataproc_jupyter_plugin import credentials
+from dataproc_jupyter_plugin.services.vertex import vertex
+
+
+class SubNetworkController(APIHandler):
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/controllers/vertex/subNetwork.py]: Move to compute.py

File: dataproc_jupyter_plugin/controllers/vertex/uiConfig.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+import aiohttp
+import tornado
+from jupyter_server.base.handlers import APIHandler
+
+from dataproc_jupyter_plugin import credentials
+from dataproc_jupyter_plugin.services.vertex import vertex
+
+
+class UIConfigController(APIHandler):
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/controllers/vertex/uiConfig.py]: Move to vertex.py

File: dataproc_jupyter_plugin/handlers.py [hunk 7]
```diff
@@ -209,6 +217,12 @@ def full_path(name):
         "bigQueryPreview": bigquery.PreviewController,
         "bigQueryProjectsList": bigquery.ProjectsController,
         "bigQuerySearch": bigquery.SearchController,
+        "api/vertex/uiConfig": uiConfig.UIConfigController,
+        "api/vertex/region": region.RegionController,
+        "api/vertex/network": network.NetworkController,
+        "api/vertex/subNetwork": subNetwork.SubNetworkController,
+        "api/vertex/sharedNetwork": sharedNetwork.SharedNetworkController,
+        "api/vertex/cloudStorageBucket": cloudStorageBucket.CloudStorageBucketController
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/handlers.py]: change this to api/storage/listBuckets

File: dataproc_jupyter_plugin/handlers.py [hunk 8]
```diff
@@ -209,6 +217,12 @@ def full_path(name):
         "bigQueryPreview": bigquery.PreviewController,
         "bigQueryProjectsList": bigquery.ProjectsController,
         "bigQuerySearch": bigquery.SearchController,
+        "api/vertex/uiConfig": uiConfig.UIConfigController,
+        "api/vertex/region": region.RegionController,
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/handlers.py]: change this to api/compute/**

**Review by harsha-accenture — CHANGES_REQUESTED** [review]

**Review by brian-nguyen — APPROVED** [review]

File: dataproc_jupyter_plugin/services/compute.py [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+        return {
+            "Content-Type": CONTENT_TYPE,
+            "Authorization": f"Bearer {self._access_token}",
+        }
+
+    async def list_region(self):
+        try:
+            regions = []
+            credentials = oauth2.Credentials(token=self._access_token)
+            compute_client = compute_v1.RegionsClient(credentials=credentials)
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/services/compute.py]: Rename the variable to regions_client

File: dataproc_jupyter_plugin/services/compute.py [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+        except Exception as e:
+            self.log.exception(f"Error fetching regions: {str(e)}")
+            return {"Error fetching regions": str(e)}
+
+    async def get_network(self):
+        try:
+            networks = []
+            credentials = oauth2.Credentials(token=self._access_token)
+            compute_client = compute_v1.NetworksClient(credentials=credentials)
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/services/compute.py]: Rename the variable to networks_client

File: dataproc_jupyter_plugin/services/compute.py [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+
+        except Exception as e:
+            self.log.exception(f"Error fetching network: {str(e)}")
+            return {"Error fetching network": str(e)}
+
+    async def get_subnetwork(self, region_id, network_id):
+        try:
+            sub_networks = []
+            credentials = oauth2.Credentials(token=self._access_token)
+            compute_client = compute_v1.SubnetworksClient(credentials=credentials)
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/services/compute.py]: Rename the variable to subnetworks_client

File: dataproc_jupyter_plugin/services/compute.py [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+
+        except Exception as e:
+            self.log.exception(f"Error fetching sub network: {str(e)}")
+            return {"Error fetching sub network": str(e)}
+
+    async def get_shared_network(self, project_id, region_id):
+        try:
+            shared_networks = []
+            credentials = oauth2.Credentials(token=self._access_token)
+            compute_client = compute_v1.SubnetworksClient(credentials=credentials)
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/services/compute.py]: Rename the variable to subnetworks_client

File: dataproc_jupyter_plugin/services/iam.py [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+            self.log.exception("Missing required credentials")
+            raise ValueError("Missing required credentials")
+        self._access_token = credentials["access_token"]
+        self.project_id = credentials["project_id"]
+        self.region_id = credentials["region_id"]
+
+    async def list_service_account(self):
+        try:
+            credentials = oauth2.Credentials(self._access_token)
+            iam_admin_client = iam_admin_v1.IAMAsyncClient(credentials=credentials)
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/services/iam.py]: Rename the variable to iam_client

File: dataproc_jupyter_plugin/controllers/storage.py [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+from dataproc_jupyter_plugin import credentials
+from dataproc_jupyter_plugin.services import storage
+
+
+class CloudStorageController(APIHandler):
+    @tornado.web.authenticated
+    async def get(self):
+        """Returns cloud storage bucket"""
+        try:
+            async with aiohttp.ClientSession() as client_session:
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/controllers/storage.py]: If we are using client library, why do we need this session?

File: dataproc_jupyter_plugin/services/storage.py [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.cloud import storage
+
+
+class Client:
+    def __init__(self, credentials, log, client_session):
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/services/storage.py]: Remove client_session if we are using client library

File: dataproc_jupyter_plugin/services/storage.py [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+            ("access_token" in credentials)
+            and ("project_id" in credentials)
+            and ("region_id" in credentials)
+        ):
+            self.log.exception("Missing required credentials")
+            raise ValueError("Missing required credentials")
+        self._access_token = credentials["access_token"]
+        self.project_id = credentials["project_id"]
+        self.region_id = credentials["region_id"]
+        self.client_session = client_session
```
Comment (harsha-accenture) [inline dataproc_jupyter_plugin/services/storage.py]: Remove client_session if we are using client library

**Review by harsha-accenture — CHANGES_REQUESTED** [review]

**Review by harsha-accenture — APPROVED** [review]

**Pull request merged**